ops.go

```diff
@@ -182,6 +185,8 @@ func (o *ImageOps) Transform(d Decoder, opt *ImageOptions, dst []byte) ([]byte,
 		// content == nil and err == nil -- this is encoder telling us to do another frame
 
 		// for mulitple frames/gifs we need the decoded frame to be active again
-		o.swap()
+		if swapped {
```

Comment:
this new boolean doesn't feel like the right way to do this, since returning only err from o.fit and o.resize was enough.  Why don't you check opt.ResizeMethod here instead?

Reply:
just talked irl with Brian and I think the way it is done in the PR makes sense -- the `fit` and `resize` methods are the ones that know whether they swapped buffers or not